src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,321 @@
+                System.out.println("Deletion cancelled.");
+            }
+            countTaskNumber();
+        } catch (Exception exception) {
+            System.out.println("This task doesn't exist!");
+        }
+    }
+
+    private static void echo(String text) {
+        String command = (!text.contains(" ")) ? text : text.substring(0, text.indexOf(" "));
+        try {
+            switch (command) {
+            case "list":
+                list();
+                break;
+            case "joke":
+                joke();
+                break;
+            case "bye":
+                exit();
+                break;
+            case "done":
+                markAsDone(text.substring(4));
+                saveFile();
+                break;
+            case "delete":
+                deleteTask(text.substring(6));
+                saveFile();
+                break;
+            case "todo":
+                addToDo(text.substring(4), true);
+                saveFile();
+                break;
+            case "deadline":
+                addDeadline(text.substring(8), true);
+                saveFile();
+                break;
+            case "event":
+                addEvent(text.substring(5), true);
+                saveFile();
+                break;
+            default:
+                throw new InvalidCommandException(0);
+            }
+        } catch (InvalidCommandException exception) {
+            switch (exception.exceptionCode) {
+            case 0:
+                System.out.println("No such command!");
+                break;
+            case 1:
+                System.out.println("Incorrect command line argument(s).\ntodo -taskName");
+                break;
+            case 2:
+                System.out.println("Incorrect command line argument(s).\ndeadline -taskName /by -taskTime");
+                break;
+            case 3:
+                System.out.println("Incorrect command line argument(s).\nevent -taskName /at -taskTime");
+                break;
+            case 4:
+                System.out.println("Incorrect command line argument(s).\ndone -taskIndex");
+                break;
+            case 5:
+                System.out.println("Incorrect command line argument(s).\ndelete -taskIndex");
+                break;
+            }
+
+        }
+        drawHorizontalLine(40);
+    }
+
+    private static void list() {
+        if (tasks.size() > 0) {
+            System.out.println("Here are the tasks in your list:");
+            for (int i = 0; i < tasks.size(); i++) {
+                System.out.println(String.format("%d.%s", i + 1, tasks.get(i)));
+            }
+        } else {
+            System.out.println("You don't have any tasks in your list now.");
+        }
+    }
+
+    private static void joke() {
+        System.out.println("There are only 10 kinds of people in this world:\nthose who know binary and those who don't.");
+    }
+
+    private static void saveFile() {
```

Comment:
I noticed that Duke.java is pretty long and clunky. I would suggest extracting methods out further. For example, you can have a new class that takes care of purely file operations such as `saveFile()`and `loadFile()`.